apispec/ext/marshmallow/__init__.py

```diff
@@ -38,6 +38,10 @@ class UserSchema(Schema):
 from .openapi import OpenAPIConverter
 
 
+def resolver(schema):
+    return schema.__name__.replace('Schema', '')
+
```

Comment:
Or

```py
    return schema.__name__.rstrip('Schema') or schema.__name__
```

`PetSchemaSchema` -> `PetSchema`
`SchemaSchema` -> `Schema`
`Schema` -> `Schema`

There would still be a collision between `SchemaSchema` and `Schema` but admittedly it doesn't matter.

Reply:
My first thought was `schema.__name__.replace('Schema', '', 1)` but this will replace by the left.

Generally, schemas are named `SomethingSchema` so it should be by the right.

Say you have a file for pet schemas, the schema for this file is `PetSchemaFileSchema`.

More likely, you want to serialize schematics, the schema is `SchematicsSchema` so we'd better strip right.

If we make this default, we should add a not in the docs to tell users to name their schemas `SomethingSchema` (or specify another resolver).

Reply:
rstrip keeps on replacing characters until it finds on not in `chars` so SchemaSchema -> ''.  Changed to only replace 'Schema' once.

Reply:
Oh, right. I was wrong about `rstrip`.

But `schema.__name__.replace('Schema', '', 1) or schema.__name__` fails on the `SchematicsSchema` case.

    'SchematicsSchema'.replace('Schema', '', 1)
    'ticsSchema'

We might have to go with a longer form:

```py
def resolve(schema):
    return schema.__name__[:-6] or schema.__name__ if schema.__name__.endswitdh('Schema') else schema.__name__
```

Reply:
Should be good now.